### PR TITLE
fix: NewRouter for unstable_defineRouter

### PR DESCRIPTION
**Pull request by dai-shi**
Previously with current Router, this issue wasn't revealed because of other bugs (?), but this change should be reasonable. (I'm not happy with the current overall implementation. Hope to revisit.)

### CODE REVIEW

File: packages/waku/src/router/client.ts [hunk 1]
```diff
@@ -808,7 +808,8 @@ export function NewRouter({
               // FIXME this check here seems ad-hoc (less readable code)
               if (
                 window.location.pathname !== path ||
-                window.location.search.replace(/^\?/, '') !== query
+                (!isStatic &&
+                  window.location.search.replace(/^\?/, '') !== query)
```
Comment (tylersayshi) [inline packages/waku/src/router/client.ts:812]: are we sure the `.replace` is still needed?

Comment (tylersayshi) [reply packages/waku/src/router/client.ts:812]: nvm I just doubled checked... it is needed

Comment (dai-shi) [reply packages/waku/src/router/client.ts:812]: I think `window.location.search` has `?`, while `query` does not, no?

Comment (tylersayshi) [reply packages/waku/src/router/client.ts:812]: you're right. I was mistaken with my initial comment, please disregard.

**Pull request merged**